File: Patterns/Abstract Factory Pattern/CSharp/PizzaTestDrive.cs
using System;

public class PizzaTestDrive
{
    public static void Main(string[] args)
    {
        PizzaStore.PizzaStore nyStore = new PizzaStore.NYPizzaStore();
        PizzaStore.PizzaStore chicagoStore = new PizzaStore.ChicagoPizzaStore();

        Pizza.Pizza? pizza = nyStore.OrderPizza("cheese");
        Console.WriteLine($"에단이 주문한 {pizza?.Name}\n");

        pizza = chicagoStore.OrderPizza("cheese");
        Console.WriteLine($"조엘이 주문한 {pizza?.Name}");
    }
}

File: Patterns/Abstract Factory Pattern/Cpp/PizzaTestDrive.cpp
#include <iostream>

#include "Pizza/Pizza.cpp"

#include "PizzaStore/PizzaStore.cpp"
#include "PizzaStore/NYPizzaStore.cpp"
#include "PizzaStore/ChicagoPizzaStore.cpp"

using namespace std;

int main() {
    PizzaStore* nyStore = new NYPizzaStore();
    PizzaStore* chicagoStore = new ChicagoPizzaStore();

    Pizza* pizza = nyStore->orderPizza("veggie");
    cout << "에단이 주문한 " << pizza->getName() << '\n';

    pizza = chicagoStore->orderPizza("cheese");
    cout << "조엘이 주문한 " << pizza->getName() << '\n';
}